### PR TITLE
[bigquery] Add scaffolding for the Storage Write API

### PR DESCRIPTION
**Pull request by nathan-artie**
Breaking out some parts of the Storage Write API PR I'm working out.

### CODE REVIEW

File: clients/bigquery/cast.go [hunk 1]
```diff
@@ -104,3 +90,22 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
 	slog.Error("Unexpected BigQuery Data Type", slog.Any("colKind", colKind.KindDetails.Kind), slog.Any("colVal", colVal))
 	return fmt.Sprint(colVal), nil
 }
+
+// EncodeStructToJSONString takes a struct as either a string or Go object and encodes it into a JSON string.
+// Structs from relational and Mongo are different.
+// MongoDB will return the native objects back such as `map[string]any{"hello": "world"}`
+// Relational will return a string representation of the struct such as `{"hello": "world"}`
+func EncodeStructToJSONString(value any) (string, error) {
```
Comment (nathan-artie) [inline clients/bigquery/cast.go:98]: Will need this for the Storage Write API.

File: clients/bigquery/cast.go [hunk 2]
```diff
@@ -104,3 +90,22 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
...
+	if colValString, isOk := value.(string); isOk {
+		if strings.Contains(colValString, constants.ToastUnavailableValuePlaceholder) {
+			return fmt.Sprintf(`{"key":"%s"}`, constants.ToastUnavailableValuePlaceholder), nil
+		}
+		return colValString, nil
+	}
+
+	colValBytes, err := json.Marshal(value)
```
Comment (nathan-artie) [inline clients/bigquery/cast.go:106]: @Tang8330 if `value` isn't a string do we care if the toast placeholder appears within it (e.g. `["__debezium_unavailable_value"]`, `{"foo": "__debezium_unavailable_value"}`)?

File: lib/config/bigquery.go [hunk 3]
```diff
@@ -5,11 +5,12 @@ import "fmt"
...
-	DefaultDataset    string `yaml:"defaultDataset"`
-	ProjectID         string `yaml:"projectID"`
-	Location          string `yaml:"location"`
-	BatchSize         int    `yaml:"batchSize"`
+	PathToCredentials  string `yaml:"pathToCredentials"`
+	DefaultDataset     string `yaml:"defaultDataset"`
+	ProjectID          string `yaml:"projectID"`
+	Location           string `yaml:"location"`
+	BatchSize          int    `yaml:"batchSize"`
+	UseStorageWriteAPI bool   `yaml:"__useStorageWriteAPI"` // Not officially supported yet.
```
Comment (nathan-artie) [inline lib/config/bigquery.go:13]: For config settings that are not intended for production use I think it's nice to prefix them with "__" to indicate that the are not officially supported, that way we can get rid of them without breaking backwards compatibility and having to bump the version.

File: clients/bigquery/cast.go (hunk 2 above)
Comment (Tang8330) [reply clients/bigquery/cast.go:106]: I don't think it's possible, but can we add a guardrail for it and Sentry when it does happen?

File: lib/config/bigquery.go (hunk 3 above)
Comment (Tang8330) [reply lib/config/bigquery.go:13]: Sure

File: clients/bigquery/cast.go [hunk 4]
```diff
@@ -96,16 +96,21 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
...
-	colValBytes, err := json.Marshal(value)
+	bytes, err := json.Marshal(value)
 	if err != nil {
 		return "", fmt.Errorf("failed to marshal colVal: %w", err)
 	}
-	return string(colValBytes), nil
+
+	stringValue := string(bytes)
+	if strings.Contains(stringValue, constants.ToastUnavailableValuePlaceholder) {
+		slog.Error("encoded JSON value contains the toast unavailable value placeholder")
```
Comment (Tang8330) [inline clients/bigquery/cast.go:113]: This should return this too

```go
return fmt.Sprintf(`{"key":"%s"}`, constants.ToastUnavailableValuePlaceholder), nil
```

**Pull request merged**